### PR TITLE
Fix issues identified with -Wall -pedantic in gcc/clang

### PR DESCRIPTION
**Pull request by ajtribick**
Fix various issues identified by running gcc 13.0.1 and clang 16.0.0 with the -Wall -pedantic options

Not enabling them in the cmake file for now because of warnings generated in the Eigen headers

### CODE REVIEW

File: src/celestia/url.cpp [hunk 1]
```diff
@@ -287,9 +287,9 @@ Url::goTo()
     // Position and orientation stored in frame coordinates; convert them
     // to universal and set the observer position.
     double tdb = sim->getTime();
-    auto coord = sim->getObserver().getFrame()->convertToUniversal(m_state.m_observerPosition, m_state.m_tdb);
+    auto coord = sim->getObserver().getFrame()->convertToUniversal(m_state.m_observerPosition, tdb);
```
Comment (ajtribick) [inline src/celestia/url.cpp:290]: Original use of `m_state.m_tdb` here was probably a bug?

File: src/celengine/shadermanager.cpp [hunk 2]
```diff
@@ -477,6 +479,7 @@ Sh_Expression operator*(const Sh_Expression& left, const Sh_Expression& right)
     return makeExpression<Sh_MultiplicationExpression>(left, right);
 }
 
+[[maybe_unused]]
```
Comment (ajtribick) [inline src/celengine/shadermanager.cpp:482]: Leaving this in the codebase for completeness with the other arithmetic operators

File: src/celengine/shadermanager.cpp [hunk 3]
```diff
@@ -598,6 +576,7 @@ Sh_Expression mix(const Sh_Expression& v0, const Sh_Expression& v1, const Sh_Exp
     return makeExpression<Sh_TernaryFunctionExpression>("mix", v0, v1, alpha);
 }
 
+[[maybe_unused]]
```
Comment (ajtribick) [inline src/celengine/shadermanager.cpp:579]: Leaving this in the codebase to complement the `max` function

File: src/celengine/render.cpp [hunk 4]
```diff
@@ -4613,17 +4609,15 @@ bool Renderer::captureFrame(int x, int y, int w, int h, PixelFormat format, unsi
...
         int realWidth = w * formatWidth(format);
         realWidth = (realWidth + 3) & ~0x3;
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-        uint8_t tempLine[realWidth]; // G++ supports VLA as an extension
-#else
-        uint8_t *tempLine = static_cast<uint8_t*>(alloca(realWidth));
-#endif
-        uint8_t *fb = buffer;
+
+        std::vector<std::uint8_t> tempLine(static_cast<std::size_t>(realWidth), 0);
```
Comment (375gnu) [inline src/celengine/render.cpp:4613]: why?

File: src/celengine/shadermanager.cpp [hunk 5]
```diff
@@ -2501,7 +2474,7 @@ ShaderManager::buildFragmentShader(const ShaderProperties& props)
     {
         if (useSeparateDiffuse)
         {
-            for (unsigned k = 0; k < props.nLights - 1; k++)
+            for (unsigned k = 0; (k + 1) < props.nLights; ++k)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:2477]: why?

File: src/celephem/vsop87.cpp [hunk 6]
```diff
@@ -11026,8 +11026,7 @@ class VSOP87Orbit : public CachingOrbit
         vsR(_vsR.data()), nR(_vsR.size()),
         period(_period),
         boundingRadius(_boundingRadius)
-    {
-    };
+    {}
```
Comment (375gnu) [inline src/celephem/vsop87.cpp:11029]:
```
{
}
```

File: src/celephem/vsop87.cpp [hunk 7]
```diff
@@ -11137,8 +11136,7 @@ class VSOP87OrbitRect : public CachingOrbit
         vsZ(_vsZ.data()), nZ(_vsZ.size()),
         period(_period),
         boundingRadius(_boundingRadius)
-    {
-    };
+    {}
```
Comment (375gnu) [inline src/celephem/vsop87.cpp:11139]:
```
{
}
```

File: src/celestia/url.cpp (hunk 1 above)
Comment (375gnu) [reply src/celestia/url.cpp:290]: yes, it should be used only when m_timeSource == UseUrlTime

File: src/celscript/lua/celx_celestia.cpp [hunk 8]
```diff
@@ -2246,6 +2246,7 @@ static int celestia_verbosity(lua_State* l)
 }
 
 
+#ifdef USE_MINIAUDIO
```
Comment (375gnu) [inline src/celscript/lua/celx_celestia.cpp:2249]: this should be inside the function body

File: src/celttf/truetypefont.cpp [hunk 9]
```diff
@@ -324,28 +319,29 @@ TextureFontPrivate::toPos(FT_ULong ch) const
     std::size_t pos = 0;
 
     if (ch > m_unicodeBlocks.back().last)
-        return -1;
+        return static_cast<std::size_t>(-1);
```
Comment (375gnu) [inline src/celttf/truetypefont.cpp:322]: it seams it worth to introduce a constant for this

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celscript/lua/celx_celestia.cpp (hunk 8 above)
Comment (ajtribick) [reply src/celscript/lua/celx_celestia.cpp:2249]: The warning here is "unused function". This function is only called when `USE_MINIAUDIO` is enabled.

File: src/celengine/shadermanager.cpp (hunk 5 above)
Comment (ajtribick) [reply src/celengine/shadermanager.cpp:2477]: `unsigned short` promotes to `int` in arithmetic operations, which then causes a signed vs unsigned comparison.

Looking at this further I'm not sure the statement generated here actually does anything?

File: src/celscript/lua/celx_celestia.cpp (hunk 8 above)
Comment (375gnu) [reply src/celscript/lua/celx_celestia.cpp:2249]: yes, correct. due to its name i decided that it's part of celx.

File: src/celengine/shadermanager.cpp [hunk 10]
```diff
@@ -558,36 +561,11 @@ Sh_Expression vec2(const Sh_Expression& x, const Sh_Expression& y)
     return makeExpression<Sh_BinaryFunctionExpression>("vec2", x, y);
 }
 
-Sh_Expression vec3(const Sh_Expression& x, const Sh_Expression& y, const Sh_Expression& z)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:561]: please leave it we can use it later (at least i plan this)

File: src/celengine/shadermanager.cpp [hunk 11]
```diff
@@ -558,36 +561,11 @@ Sh_Expression vec2(const Sh_Expression& x, const Sh_Expression& y)
...
-{
-    return makeExpression<Sh_TernaryFunctionExpression>("vec3", x, y, z);
-}
-
 Sh_Expression dot(const Sh_Expression& v0, const Sh_Expression& v1)
 {
     return makeExpression<Sh_BinaryFunctionExpression>("dot", v0, v1);
 }
 
-Sh_Expression cross(const Sh_Expression& v0, const Sh_Expression& v1)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:571]: please leave it we can use it later (at least i plan this)

File: src/celengine/shadermanager.cpp [hunk 12]
```diff
@@ -558,36 +561,11 @@ Sh_Expression vec2(const Sh_Expression& x, const Sh_Expression& y)
...
-{
-    return makeExpression<Sh_BinaryFunctionExpression>("cross", v0, v1);
-}
-
-Sh_Expression sqrt(const Sh_Expression& v0)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:576]: please leave it we can use it later (at least i plan this)

File: src/celengine/shadermanager.cpp [hunk 13]
```diff
@@ -558,36 +561,11 @@ Sh_Expression vec2(const Sh_Expression& x, const Sh_Expression& y)
...
-{
-    return makeExpression<Sh_UnaryFunctionExpression>("sqrt", v0);
-}
-
-Sh_Expression length(const Sh_Expression& v0)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:581]: please leave it we can use it later (at least i plan this)

File: src/celengine/shadermanager.cpp [hunk 14]
```diff
@@ -558,36 +561,11 @@ Sh_Expression vec2(const Sh_Expression& x, const Sh_Expression& y)
...
-{
-    return makeExpression<Sh_UnaryFunctionExpression>("length", v0);
-}
-
-Sh_Expression normalize(const Sh_Expression& v0)
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:586]: please leave it we can use it later (at least i plan this)

File: src/celengine/shadermanager.cpp [hunk 15]
```diff
@@ -702,12 +681,6 @@ assign(std::string_view variableName, const Sh_Expression& expr)
     return fmt::format("{} = {};\n", variableName, expr.toString());
 }
 
-std::string
```
Comment (375gnu) [inline src/celengine/shadermanager.cpp:705]: please leave it we can use it later (at least i plan this)

File: src/celestia/view.cpp [hunk 16]
```diff
@@ -262,8 +262,8 @@ void View::drawBorder(int gWidth, int gHeight, const Color &color, float linewid
 
 void View::updateFBO(int gWidth, int gHeight)
```
Comment (375gnu) [inline src/celestia/view.cpp:263]: maybe better to change signature of this method to accept `unsigned`? i can't imaging negative dimensions

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celestia/view.cpp (hunk 16 above)
Comment (ajtribick) [reply src/celestia/view.cpp:263]: Probably, but doing that ends up with knock-on changes to celestiacore.

**Review by 375gnu — APPROVED** [review]

**Pull request merged**